BoxContentSDK/BoxContentSDK/Requests/BOXItemSetCollectionsRequest.m

```diff
@@ -50,53 +78,122 @@ - (instancetype)initFolderSetCollectionsRequestForFolderWithID:(NSString *)folde
-    
-    NSURL *url = [self URLWithResource:self.resource 
+    NSURL *url = [self URLWithResource:self.resource
                                     ID:self.itemID
                            subresource:nil
                                  subID:nil];
-
+    
     NSDictionary *queryParameters = nil;
     if (self.requestAllItemFields) {
         queryParameters = @{BOXAPIParameterKeyFields :[self fullItemFieldsParameterString]};
     }
-
+    
     NSMutableArray *bodyContent = [NSMutableArray array];
+    
     for (NSString *collectionID in self.collectionIDs) {
         [bodyContent addObject:@{BOXAPIObjectKeyID : collectionID}];
     }
     NSDictionary *bodyDictionary = @{BOXAPIObjectKeyCollections : bodyContent};
     
-    operation = [self JSONOperationWithURL:url 
-                                HTTPMethod:BOXAPIHTTPMethodPUT
-                     queryStringParameters:queryParameters
-                            bodyDictionary:bodyDictionary
-                          JSONSuccessBlock:nil
-                              failureBlock:nil];
-    
-    return operation;
+    if (self.associateId != nil) {
+        
+        BOXAPIDataOperation *dataOperation = [self dataOperationWithURL:url
+                                                             HTTPMethod:BOXAPIHTTPMethodPUT
+                                                  queryStringParameters:queryParameters
+                                                         bodyDictionary:bodyDictionary
+                                                           successBlock:nil
+                                                           failureBlock:nil
+                                                            associateId:self.associateId];
+        
+        if (self.associateId != nil) {
+             NSString *destinationPath = [NSTemporaryDirectory() stringByAppendingPathComponent:self.associateId];
+            dataOperation.destinationPath = destinationPath;
+        }
+        
+        return dataOperation;
+    } else {
+        BOXAPIJSONOperation *operation = [self JSONOperationWithURL:url
+                                                         HTTPMethod:BOXAPIHTTPMethodPUT
+                                              queryStringParameters:queryParameters
+                                                     bodyDictionary:bodyDictionary
+                                                   JSONSuccessBlock:nil
+                                                       failureBlock:nil];
+        
+        return operation;
+    }
 }
 
 - (void)performRequestWithCompletion:(BOXItemBlock)completionBlock
 {
-    BOOL isMainThread = [NSThread isMainThread];
-    BOXAPIJSONOperation *collectionAddItemOperation = (BOXAPIJSONOperation *)self.operation;
-    
-    collectionAddItemOperation.success = ^(NSURLRequest *request, NSHTTPURLResponse *response, NSDictionary *JSONDictionary) {
-
+    if (self.associateId != nil) {
+        if (completionBlock) {
```

Comment:
should execute even if completionBlock is nil because we could still cache the response